include/onnxruntime/core/providers/dml/dml_provider_factory.h

```diff
@@ -5,7 +5,13 @@
 
 #pragma warning(push)
 #pragma warning(disable : 4201) // nonstandard extension used: nameless struct/union
+#ifdef _GAMING_XBOX_SCARLETT
+#include <d3d12_xs.h>
+#elif _GAMING_XBOX_XBOXONE
```

Comment:
nit: mixing `#ifdef` and `#elif` - should this be `#if`, or `#elif defined()`?

Reply:
C++23 will have `elifdef`, which maybe we can use in 5 years :b.